fix(notifications): fix pointer events for inline low contrast, update inverse link colors

jendowns opened the pull request:
Closes #4804 

This PR fixes the pointer events issue identified in #4804 and seen on the [`gatsby-theme-carbon` site](https://gatsby-theme-carbon.now.sh/components/InlineNotification) by adding `pointer-events: none` to the pseudo-selector `:before` styles causing the problem.

I also updated the storybook to add example links to both Notification components' `subtitle` prop, so that links can easily be checked in the future.

Also in the process of doing this, I noticed that links passed into `title` or `subtitle` props aren't styled correctly with the `$inverse-link` token, so I went ahead and added styles for that. **I'm just not sure what hover/focus color you want for those links?** Just let me know & I can update it 👍 

#### Changelog

**New**

- use `$inverse-link` color tokens for links inside `InlineNotification` and `ToastNotification`

**Changed**

- for low contrast `InlineNotification`, add `pointer-events:none` to the pseudo `:before` that covers the component
- add link example to both notification components in storybook

#### Testing / Reviewing

Make sure to select the `Low Contrast` knob in storybook. I definitely suggest comparing the deploy with `InlineNotification` usage on the  [`gatsby-theme-carbon` site](https://gatsby-theme-carbon.now.sh/components/InlineNotification) 👍 

Also after you toggle on `Low Contrast` knob in this PR's storybook deploy, you can inspect the low-contrast `:before` psuedo selector & comment out the `pointer-events: none` style, then see if you can click on the link inside the `InlineNotification` (you shouldn't be able to)

jendowns committed:
fix(inline-notification): update low contrast pointer event, link color

jendowns committed:
fix(toast-notification): update inverse link color

jendowns committed:
docs(notifications): update stories to have example links

jendowns committed:
Merge branch 'master' into 4804_notification-low-contrast

abbeyhrt commented on packages/react/src/components/Notification/Notification-story.js:
```diff
@@ -26,7 +26,11 @@ const notificationProps = () => ({
-  subtitle: text('Subtitle (subtitle)', 'Subtitle text goes here.'),
+  subtitle: (
+    <span>
+      Subtitle text goes here. <a href="#example">Example link</a>
```
I'm not sure if this will fix it but the `a` tag here is causing a color contrast DAP violation in low contrast mode, would switching it to the `Link` component fix it? 
```suggestion
      Subtitle text goes here. <Link href="#example">Example link</Link>
```

abbeyhrt replied:
Ignore that! It doesn't resolve the violation.

jendowns replied:
Thanks @abbeyhrt! I was thinking in this case that a user may be likely to just pass in a plain anchor element instead of an actual Link component, since the prop is a `node` type.

About the colors... I went with `$inverse-link` because the token seemed appropriate, but I can definitely adjust to something lighter? 🤔 I'm also unsure what the color of the link should be when you hover/focus on it... as it is, I can't necessarily leave it as-is because the component is inverted so any anchor/`Link` passed it will be even more unreadable (like blue60 by default 😅 )

What do you think about the colors though? Lighter like Blue30 or something like that?

abbeyhrt replied:
Ah that makes sense! I'm finding the color contrast violation only exists in low contrast mode in the white and g10 themes, so I would think we would need a darker value? But since it takes on the theme value, it can't be too dark or it causes a violation in the default mode 🤦‍♀ .  I'm thinking we would need to do something special for links when low contrast mode is on.  

For the hover, maybe we could follow the pattern 'Link' has? Toggling the underline on hover.

jendowns committed:
Merge branch 'master' into 4804_notification-low-contrast

joshblack commented:
One thing that @dakahn pointed out during our PR review was that the links seemingly don't hit color contrast, did you notice this too @jendowns?

![image](https://user-images.githubusercontent.com/3901764/70268865-dff99400-1766-11ea-9ef9-db029a8f556c.png)

jendowns committed:
Merge branch 'master' into 4804_notification-low-contrast

jendowns committed:
fix(notifications): update inverse link color selectors

jendowns committed:
fix(notifications): add styles for lines inside notifications

jendowns commented:
Thanks for your notes on the colors @abbeyhrt & @joshblack! I updated the `inverse-link` color rules I added to *only* apply to the non-low-contrast version of the components.

I also went ahead and added some hover and focus styles for links inside notifications... I realize it would be ideal if people passed in the `Link` component in this situations, but I think it's also super common for someone to just pass in an plain anchor. Let me know what you think!

jendowns committed:
Merge branch 'master' into 4804_notification-low-contrast

jendowns committed:
Merge branch 'master' into 4804_notification-low-contrast

jendowns committed:
Merge branch 'master' into 4804_notification-low-contrast

jendowns committed:
Merge branch 'master' into 4804_notification-low-contrast

abbeyhrt reviewed (approved):
This looks good to me! Thank you Jen!!

jendowns committed:
Merge branch 'master' into 4804_notification-low-contrast

aledavila reviewed (approved):
lgtm

aagonzales reviewed (approved):
Ok I think this looks good. No color change with just the added underline on hover is good.

joshblack committed:
Merge branch 'master' into 4804_notification-low-contrast

joshblack committed:
Merge branch 'master' into 4804_notification-low-contrast
